File: src/pages/blog.js
import React from "react"
import { Link } from "gatsby"
import { graphql } from "gatsby"

import Layout from "../components/layout"
import SEO from "../components/seo"

const BlogPage = ({ data }) => {
  return (
    <Layout>
      <SEO title="Blog" />
      <h1>Latest Posts</h1>
      {data.allMarkdownRemark.edges.map(post => {
        const { id } = post.node
        const { title, summary, path, author, date } = post.node.frontmatter

        return (
          <div className="card" key={id}>
            <div className="container">
              <h3>
                <b>{title}</b>
              </h3>
              <hr />
              <p>{summary}</p>
              <br />
              <div className="card-bottom">
                <Link className="btn btn-primary" to={path}>
                  Read More
                </Link>
                <small className="author">
                  {author} on {date}
                </small>
              </div>
            </div>
          </div>
        )
      })}
    </Layout>
  )
}

export const pageQuery = graphql`
  query BlogIndexQuery {
    allMarkdownRemark(sort: { fields: [frontmatter___date], order: DESC }) {
      edges {
        node {
          id
          frontmatter {
            path
            date
            title
            summary
            author
          }
        }
      }
    }
  }
`

export default BlogPage

File: python_script/blog.py
class Blog:
  def __init__(self, path, date, title, author, content, language, code=''):
    self.path = path
    self.date = date
    self.title = title
    self.author = author
    self.content = content
    if language == 'Choose a programming language':
      language = None
    self.language = language
    self.code = code.strip()
    print('--> code = ' + code)

  def __str__(self):
    if self.language is not None:
      return f'''---
path: {self.path}
date: {self.date}
title: {self.title}
author: {self.author}
---

{self.content}

```{self.language}
{self.code}
```
'''
    else:
      return f'''---
path: {self.path}
date: {self.date}
title: {self.title}
author: {self.author}
---

{self.content}
'''

